tests/e2e/framework/common.go

```diff
@@ -800,6 +874,22 @@ func (f *Framework) createInvalidMachineConfigPool(n string) error {
 		ObjectMeta: metav1.ObjectMeta{Name: n},
 		Spec: mcfgv1.MachineConfigPoolSpec{
 			Paused: false,
+			// Add minimal selectors to pass ValidatingAdmissionPolicy
+			// This pool is still "invalid" for testing as no nodes match this selector
+			NodeSelector: &metav1.LabelSelector{
```

Comment:
I had to do some digging to better understand this. It looks like we added the `e2e-invalid` pool intentionally to make sure the Compliance Operator would still successfully schedule scans, even if there was an invalid pool defined in the cluster:

4961d0ff4

If we add a node selector here, we won't be testing if this condition successfully returns `false` will we?

https://github.com/ComplianceAsCode/compliance-operator/blob/master/pkg/utils/nodeutils.go#L209-L211

Reply:
Perhaps it's not necessary to have that level of protection built into the operator on newer releases that prevent users from creating machine config pools without a node selector.